src/main/java/com/store/book/controller/BookController.java

```diff
@@ -35,7 +37,11 @@ public BookDto save(@Valid @RequestBody CreateBookRequestDto bookDto) {
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of all available books")
     public Page<BookDto> findAll(Pageable pageable) {
-        return bookService.findAll(pageable);
+        Authentication authentication = SecurityContextHolder
+                .getSecurityContext()
+                .getAuthentication();
+        String email = (String) authentication.getPrincipal();
+        return bookService.findAll(email, pageable);
```

Comment:
Why do you need it?

Reply:
We retrieve the authenticated user's email to fetch only the books related to that user. bookService.findAll(email, pageable) returns user-specific results.